src/main/java/mnsky/TaskList.java

```diff
@@ -0,0 +1,255 @@
+package mnsky;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+import mnsky.exceptions.MnskyException;
+import mnsky.exceptions.MnskyInvalidParameterException;
+import mnsky.task.Deadline;
+import mnsky.task.Event;
+import mnsky.task.Task;
+
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    /**
+     * Creates a TaskList object. Retrieves the storage data if there are no issues with it.
+     * @param ui The ui object, to be used for printing an exception.
+     * @param storage The storage object, to be used for retrieving the storage data.
+     */
+    public TaskList(Ui ui, Storage storage) {
+        tasks = new ArrayList<>();
+
+        try {
+            getStorageData(storage);
+        } catch (MnskyException e) {
+            ui.printException(e);
+            tasks = new ArrayList<>();
+        }
+    }
+
+    /**
+     * Creates a new task based on the parameters.
+     * @param name The name of the task.
+     * @return The new task.
+     */
+    public Task addTask(String name) {
+        Task task = new Task(name);
+        tasks.add(task);
```

Comment:
Perhaps renaming the ArrayList variable as "tasklist" or "list" instead of "tasks" would be clearer? 😄

Reply:
Not sure, I called it tasks instead of taskList so that it wouldn't be confused with the actual taskList object itself. The coding standard also mentioned that lists of objects should be named as plural nouns.